### ISSUE TITLE
How to define RenderValueComponent while using vanilla-jsoneditor

### ISSUE THREAD

**Issue by 1pone** [josdejong/svelte-jsoneditor#375]
I'm using vanilla-jsoneditor in React, and i need use custom RenderValueComponent  for `onRenderValue.component` to render value as password, like the ReadonlyPassword example shown in  the project. But how can i define the RenderValueComponent without svelte?

**Comment by 1pone** [comment]
Or can I submit a PR to add `ReadonlyPassword` export?

**Comment by josdejong** [comment]
I haven't tried out if/how you can use `onRenderValue` from an other framework like React, we have to do some experimentation there.

I like the idea of exporting `ReadonlyPassword`: it can be the first a collection of extra value render components. I think though that we should either rename it to something like `HiddenValue`, or improve on it: it was just a minimal example, and when you double click it to edit, you see all of the password in plain sight, that should remain hidden whilst typing.

**Comment by AlexRMU** [comment]
I think you can do it like [this](https://svelte-5-preview.vercel.app/#H4sIAAAAAAAACn1TwY6cMAz9FSvaA6hopr0yw0j9hR56aaoKgplmFxyamO2uEP9eJWGYoNGWC-C8Zz8_O7PodI9OlD9mQfWAohRfx1EUgt9H_-NesWcUhXBmsspHJJ-dsnrkiyQAAD2MxjLMYOgbUov2e91PCAt01gwgxeGoqcW3w7OT4hQpPTIoMxFDBU-Oa8bsc36SFE-7iRRrQ6BJWRyQOMthjmf-icxPFXw5pUFypsdDb66ZFBtTiiLi8xW73Kp4DaM1o_MaWrT6FdtsjuACateWIMWfvygFLHmqu4EKshyqC8zLGn_CrkPF2RpOVDVZqLFVv30kPeqH9sgxKIs1I1R7V_fZ7mhDv9Tvmq6eEJlZa9TkHTg0pn3fMZqQdSXcbTkft7FKOjcTsyEwpHqtXqo5tpZMZLmEE1fCHDxbzsfIufyHrQPRvxO0KMRgWt1pbEXJdsKl2HYxrM59G59duombiVs3qz83O_dLsaH8Uszr8MNEwlJs2VYDubZX5A9yRUxMFIH32VrkydLO4lAA38JFSVQ_TvYj5QnyQX1SM8ryBR9N_bn8AwuB7aXsAwAA) (after https://github.com/josdejong/svelte-jsoneditor/issues/373).
Other frameworks will be able to create everything they need, and reactivity both inside and outside the component will be provided by runes.

**Comment by josdejong** [comment]
Do you mean defining an API like a [Svelte action](https://svelte.dev/docs/svelte-action)?

**Comment by AlexRMU** [comment]
Yes

**Comment by josdejong** [comment]
I think that makes sense indeed.

**Comment by josdejong** [comment]
I tried out using Svelte actions, it works like a charm, see #398

**Comment by josdejong** [comment]
I've finished and merged #398 (not yet published)

**Comment by 13832198312** [comment]
<img width="414" alt="image" src="https://github.com/josdejong/svelte-jsoneditor/assets/72346243/6e44fc1c-4774-4ddf-ac7c-651e5d03c89a">
Can you give me a specific example of how to customize a component

**Comment by josdejong** [comment]
Yes, see [/examples/browser/custom_value_renderer.html](https://github.com/josdejong/svelte-jsoneditor/blob/main/examples/browser/custom_value_renderer.html)

**Comment by 13832198312** [comment]
onRenderValue={(props: any) => {
          const myRendererAction = {
            action: (node: any) => {
              const content = document.createElement('div')
              content.innerHTML = props.value
              node.classList.add('highlight')
              node.appendChild(content)
              return {
                update: (node: any) => {
                  // update the DOM
                },
                destroy: () => {
                  // cleanup the DOM
                },
              }
            },
            props,
          }
          return [myRendererAction]
        }
        return renderValue(props)
      }
     可以这样写自己的样式
